web/tests/integration/team.test.ts

```diff
@@ -11,3 +21,204 @@ describe("team model", () => {
+        update_team(where: { id: { _eq: $id } }, _set: { name: $name }) {
+          affected_rows
+        }
+      }
+    `;
+
+    for (const teamMember of teamMemberships) {
+      const client = await getAPIUserClient({
+        team_id: team_id,
+        user_id: teamMember.user_id,
+      });
+
+      const response = await client.mutate({
+        mutation,
+        variables: {
+          id: team_id,
+          name: "new name",
+        },
+      });
+
+      if (teamMember.role === Role_Enum.Owner) {
+        expect(response.data?.update_team?.affected_rows).toEqual(1);
+
+        // NOTE: revert update changes
+        await integrationDBExecuteQuery(
+          // update team hasura postgresql query
+          `UPDATE "public"."team" SET "name" = '${nameBeforeUpdate}' WHERE "id" = '${team_id}' RETURNING "name";`
+        );
+      } else {
+        expect(response.data?.update_team?.affected_rows).toEqual(0);
+        // Additional SQL query to confirm the team name has not changed
+        const { rows: unchangedTeam } = (await integrationDBExecuteQuery(
+          `SELECT name FROM "public"."team" WHERE "id" = '${team_id}'`
+        )) as { rows: Array<{ name: string }> };
+        expect(unchangedTeam[0].name).toEqual(nameBeforeUpdate);
+      }
+    }
+  });
+
+  test("cannot select another team", async () => {
+    const { rows: teams } = (await integrationDBExecuteQuery(
+      `SELECT id FROM "public"."team"`
+    )) as { rows: Array<{ id: string }> };
+
+    for (const team of teams) {
+      const { rows: teamMemberships } = (await integrationDBExecuteQuery(
+        `SELECT id, user_id FROM "public"."membership" WHERE "team_id" = '${team.id}' limit 1;`
+      )) as { rows: Array<{ id: string; user_id: string }> };
+
+      const client = await getAPIUserClient({
+        team_id: team.id,
+        user_id: teamMemberships[0].user_id,
+      });
+
+      const query = gql`
+        query ListTeams {
+          team {
+            id
+          }
+        }
+      `;
+
+      const response = await client.query({ query });
+      expect(response.data.team.length).toEqual(1);
+      expect(response.data.team[0].id).toEqual(team.id);
+    }
+    // Test using an team that the user is not a member of
+    const { rows: testInvalidTeamMemberships } =
+      (await integrationDBExecuteQuery(
+        `SELECT id, user_id FROM "public"."membership" WHERE "team_id" = '${teams[1].id}' limit 1;`
+      )) as { rows: Array<{ id: string; user_id: string }> };
+
+    const testInvalidClient = await getAPIUserClient({
+      team_id: teams[0].id,
+      user_id: testInvalidTeamMemberships[0].user_id,
+    });
+
+    const query = gql`
+      query ListTeams {
+        team (where: {id: {_eq: "${teams[0].id}"}}) {
+          id
+        }
+      }
+    `;
+
+    const response = await testInvalidClient.query({ query });
+    expect(response.data.team.length).toEqual(0);
+  });
+
+  test("API Key: cannot select another team", async () => {
```

Comment:
to follow the convention above, this should be a `describe("api_key role", () => {`